src/sqlfluff/rules/L003.py

```diff
@@ -416,358 +466,220 @@ def _process_current_line(
-                    )
-                else:
-                    # We'll need this value later.
-                    this_indent_num = this_line["indent_size"] // self.tab_space_size
-
-                # We know that the indent balance is higher, what actually is
-                # the difference in indent counts? It should be a whole number
-                # if we're still here.
-                comp_indent_num = res[k]["indent_size"] // self.tab_space_size
-
-                # The indent number should be at least 1, and can be UP TO
-                # and including the difference in the indent balance.
-                if comp_indent_num == this_indent_num:
-                    # We have two lines indented the same, but with a different starting
-                    # indent balance. This is either a problem OR a sign that one of the
-                    # opening indents wasn't used. We account for the latter and then
-                    # have a violation if that wasn't the case.
-
-                    # Does the comparison line have enough unused indent to get us back
-                    # to where we need to be? NB: This should only be applied if this is
-                    # a CLOSING bracket.
-
-                    # First work out if we have some closing brackets, and if so, how
-                    # many.
-                    b_idx = 0
-                    b_num = 0
-                    while True:
-                        if len(this_line["line_buffer"][b_idx:]) == 0:
-                            break
-
-                        elem = this_line["line_buffer"][b_idx]
-                        if not elem.is_code:
+            this_indent_num = this_line.indent_size // self.tab_space_size
+            # We know that the indent balance is higher, what actually is
+            # the difference in indent counts? It should be a whole number
+            # if we're still here.
+            comp_indent_num = prev_line.indent_size // self.tab_space_size
+
+            # The indent number should be at least 1, and can be UP TO
+            # and including the difference in the indent balance.
+            if comp_indent_num == this_indent_num:
+                # We have two lines indented the same, but with a different starting
+                # indent balance. This is either a problem OR a sign that one of the
+                # opening indents wasn't used. We account for the latter and then
+                # have a violation if that wasn't the case.
+
+                # Does the comparison line have enough unused indent to get us back
+                # to where we need to be? NB: This should only be applied if this is
+                # a CLOSING bracket.
+
+                # First work out if we have some closing brackets, and if so, how
+                # many.
+                b_idx = 0
+                b_num = 0
+                while True:
+                    if len(this_line.line_buffer[b_idx:]) == 0:
```

Comment:
As I think you pointed out before, this line looks odd. It's actually slicing the array, but not really "using" the result, only checking the length. It should be possible to replace this line with something that just checks the length. Something like:
```
if len(this_line.line_buffer) < b_idx
```

(I wrote the line above without much thought, definitely double check this before using.)

Reply:
l'll refac this whole section, its basically junk :)